cmd/test_runner_test.go

```diff
@@ -246,3 +249,49 @@ func executeCommandC(root *cobra.Command, args ...string) (err error) {
 	_, err = root.ExecuteC()
 	return err
 }
+
+// AddDatasetToRefstore adds a dataset to the test runner's refstore. It ignores the upper-levels
+// of our stack, namely cmd/ and lib/, which means it can be used to add a dataset with a name
+// that is using upper-case characters.
+func (run *TestRunner) AddDatasetToRefstore(ctx context.Context, t *testing.T, refStr string, ds *dataset.Dataset) {
+	ref, err := dsref.ParseHumanFriendly(refStr)
+	if err != nil && err != dsref.ErrBadCaseName {
+		t.Fatal(err)
+	}
+
+	ds.Peername = ref.Username
+	ds.Name = ref.Name
+
+	inst, err := lib.NewInstance(ctx, run.RepoRoot.QriPath)
+	// NOTE(dustmop): There's a bug with TestRepo that I don't understand completely. The commands
```

Comment:
innnteresting. definitely please file this same comment as an issue so we can get it cleaned up